mlflow/xgboost.py

```diff
@@ -150,9 +173,14 @@ def save_model(
     # Save an XGBoost model
     xgb_model.save_model(model_data_path)
 
+    _validate_env_arguments(conda_env, pip_requirements, additional_pip_requirements)
+
     conda_env_subpath = "conda.yaml"
     if conda_env is None:
-        conda_env = get_default_conda_env()
+        final_pip_requirements = _determine_pip_requirements(
+            _get_default_pip_deps(), pip_requirements, additional_pip_requirements,
+        )
```

Comment:
What happens if there's an error in this procedure? Should we throw a clear `MlflowException` indicating what happened?

Reply:
There are a couple cases where this procedure fails:

#### 1. Invalid value is passed to `pip_requirements` or `additional_pip_requirements`

In this case, an error indicating that they only accept a string or an interable of strings, is thrown.

#### 2. Given requirements file doesn't exist

In this case, `FileNotFoundError` is thrown in `_parse_pip_requirements`, which I think is clear enough.

So I don't think we need to throw clear `MlflowException` for now.

Reply:
Sounds good!